docs/markdown/DevGuide_Data_Memory.md

```diff
@@ -0,0 +1,85 @@
+// Iterate through buffers to find the Vertex Position buffer
+auto it = std::find_if(buffers.begin(), buffers.end(), [](const auto &buff) {
+    return buff.first.type == BufferType::PositionAttrib;
+});
+// If buffer exists and is not empty
+if (it != buffers.end() && it->second->getSize() > 0) {
+    const auto minmax = it->second           // Get actual pointer to buffer
+        ->getRepresentation<BufferRAM>()     // Get RAM representation
+        ->dispatch<std::pair<dvec4, dvec4>>( // Dispatch to produce a pair of vec4s as result
+            [](auto br) { // Deduce actual data type automatically, set ValueType
+                // br's type is deduced to BufferRAMPrecision<ValueType, BufferTarget::Data>
+                // which means it holds values of type ValueType and is a Data buffer
+                // ValueType can be extracted from br's complex type like so:
+                using ValueType = util::PrecisionValueType<decltype(br)>;
+                const auto &data = br->getDataContainer(); // Gets you a std::vector<ValueType>&
+                // Construct result as pair of ValueTypes
+                using Res = std::pair<ValueType, ValueType>;
+                // Initialize result with max and min values of given data type
+                Res minmax{DataFormat<ValueType>::max(),
+                           DataFormat<ValueType>::lowest()};
+                // Iterate over buffer, accumulating the min and max components
+                minmax = std::accumulate(data.begin(), data.end(), minmax,
+                    [](const Res &mm, const ValueType &v) -> Res {
+                        // Take component-wise min and max
+                        return {glm::min(mm.first, v), glm::max(mm.second, v)};
+                });
+                // Convert to pair of dvec4, as we specified in dispatch's type parameter
+                return std::pair<dvec4, dvec4>{util::glm_convert<dvec4>(minmax.first),
+                                               util::glm_convert<dvec4>(minmax.second)};
+        });
+        // minmax now has the lower left and upper right of the meshes bounding box.
+    }
+```
+We start as expected: first the `Mesh` handle is retrieved, then we find the buffer of interest, which is the vertex position buffer in our case. If this buffer exists, we will request its `BufferRAM` representation. Since this representation already exists in an up-to-date state, it can be retrieved without any memory transfer. On this representation the `dispatch<ResultType>(Lambda)` function is called. The actual data access is limited to the Lambda that is passed to `dispatch()` and the result type must match the given type parameter. The actual computations on the data happens inside this Lambda. If any additional variables from the outside scope are required in the Lambda, they can be passed into the Lambdas scope by adding them into the brackets like so:
```

Comment:
you might note that that the lambda has to compile for all our 40 or so kinds of ValueType, int, float, vec2 dvec4 etc. as found in https://github.com/inviwo/inviwo/blob/1b65add43ce3f16f65646234529d235d811176a5/include/inviwo/core/util/formats.h#L55
This can quite often lead to some surprising issues...